internal/crypto/internal_ca.go

```diff
@@ -133,42 +128,29 @@ func signCertificate(template *x509.Certificate, requestKey crypto.PublicKey, is
 	return certs[0], nil
 }
 
-
 // IssueRequestedClientCertificate issues a client certificate based on the provided
 // Certificate Signing Request (CSR) and the desired expiration time in seconds.
 // This currently processes both enrollment cert and management cert signing requests, which both are signed
 // by the FC service's internal CA instance named 'ca'.
-func (ca *CA) IssueRequestedClientCertificate(csr *x509.CertificateRequest, expirySeconds int) ([]byte, error) {
+func (ca *CA) IssueRequestedCertificateAsX509(csr *x509.CertificateRequest, expirySeconds int, usage []x509.ExtKeyUsage) (*x509.Certificate, error) {
 	now := time.Now()
+	expire := time.Duration(expirySeconds) * time.Second
 	template := &x509.Certificate{
-		Subject: csr.Subject,
-
-		Signature:          csr.Signature,
-		SignatureAlgorithm: csr.SignatureAlgorithm,
-
-		PublicKey:          csr.PublicKey,
-		PublicKeyAlgorithm: csr.PublicKeyAlgorithm,
-
-		Issuer: ca.Config.Certs[0].Subject,
-
-		NotBefore:    now.Add(-1 * time.Second),
-		NotAfter:     now.Add(time.Duration(expirySeconds) * time.Second),
-		SerialNumber: big.NewInt(1),
-
+		Subject:               csr.Subject,
+		Signature:             csr.Signature,
+		SignatureAlgorithm:    csr.SignatureAlgorithm,
+		PublicKey:             csr.PublicKey,
+		PublicKeyAlgorithm:    csr.PublicKeyAlgorithm,
+		IPAddresses:           csr.IPAddresses,
+		DNSNames:              csr.DNSNames,
+		Issuer:                ca.Config.Certs[0].Subject,
+		NotBefore:             now.Add(-time.Second),
+		NotAfter:              now.Add(expire),
+		SerialNumber:          big.NewInt(1),
```

Comment:
Should the SerialNumber not be generated with the CA's `SerialGenerator`? I'm not very familiar with this but when I looked it up it looked like it would be relevant

Reply:
Looks like this is from pre-existing code anyway

Reply:
That is a non-issue. This is the template, not the sign. The number is correctly generated when the CA signs the cert. 
However, in the process of reviewing it I found that we are generating the CA serial itself "by hand". That is incorrect - it also needs to be generated using the serial number generator. It is the first number for a sequential or random if all certs are random.